### PR TITLE
Add mlflow server deployment template

### PR DESCRIPTION
**Pull request by 4n4nd**
This template should deploy the most recent version of mlflow server and create a route for the server as well.

### CODE REVIEW

File: mlflow-server-oc-deployment-template.yaml [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+kind: Template
+
+labels:
+  application: ${APPLICATION_NAME}
+
+metadata:
+  name: mlflow-server-deployment-template
+
+parameters:
+- description: The name for job
```
Comment (durandom) [inline mlflow-server-oc-deployment-template.yaml]: should be "Appliaction Name"

File: mlflow-server-oc-deployment-template.yaml [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+  value: 'mlflow-server-test'
```
Comment (durandom) [inline mlflow-server-oc-deployment-template.yaml]: just `mlflow-server`

File: mlflow-server-oc-deployment-template.yaml [hunk 3]
```diff
@@ -0,0 +1,124 @@
+apiVersion: v1
+kind: Template
+
+labels:
+  application: ${APPLICATION_NAME}
+
+metadata:
+  name: mlflow-server-deployment-template
```
Comment (durandom) [inline mlflow-server-oc-deployment-template.yaml]: `mlflow-server-template`, drop "deployment"

File: mlflow-server-oc-deployment-template.yaml [hunk 4]
```diff
@@ -0,0 +1,124 @@
+apiVersion: v1
```
Comment (durandom) [inline mlflow-server-oc-deployment-template.yaml]: Then name of the file should be `mlflow-server-template.yaml` and move it to a subdirectory called `openshift`, so it's consistent with our other repos, see https://github.com/AICoE/prometheus-flatliner

File: mlflow-server-oc-deployment-template.yaml [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+      limits:
+        memory: 2Gi
+        cpu: "1"
+    output:
+      to:
+        kind: ImageStreamTag
+        name: ${APPLICATION_NAME}:latest
+    source:
+      type: "Dockerfile"
+      dockerfile: "FROM docker.io/centos/python-36-centos7:latest\nRUN pip install mlflow"
```
Comment (durandom) [inline mlflow-server-oc-deployment-template.yaml]: For now this is ok, but later it should be a s2i build similar to https://github.com/AICoE/openshift_kubeflow_workshop/blob/master/openshift/build_config_training.yaml

**Comment by zmhassan** [comment]
@4n4nd All the versions of mlflow have been tagged up to the latest which is 0.8.1.

https://quay.io/repository/zmhassan/mlflow?tab=tags

**Comment by 4n4nd** [comment]
> @4n4nd All the versions of mlflow have been tagged up to the latest which is 0.8.1.
> 
> https://quay.io/repository/zmhassan/mlflow?tab=tags

Oh sorry I didn't see that.

**Comment by durandom** [comment]
@zmhassan where is the source of your mlflow image on quay? and in the long term, I'd rather have the server as an s2i build with the source here in this repo.

File: mlflow-server-oc-deployment-template.yaml (hunk 5 above)
Comment (4n4nd) [reply mlflow-server-oc-deployment-template.yaml]: I'll change this in the next commit

**Comment by zmhassan** [comment]
The mlflow jobs are using s2i to build it. Server is custom image Docker file is in this repo:

https://github.com/AICoE/experiment-tracking/blob/master/Dockerfile
You'll need to update this to the pypi release version of mlflow

Then I have a helper script that will let you build the image.

Perhaps we can automate this with quay automated docker builds.

**Comment by 4n4nd** [comment]
> The mlflow jobs are using s2i to build it. Server is custom image Docker file is in this repo:
> 
> https://github.com/AICoE/experiment-tracking/blob/master/Dockerfile
> You'll need to update this to the pypi release version of mlflow
> 
> Then I have a helper script that will let you build the image.
> 
> Perhaps we can automate this with quay automated docker builds.

@zmhassan I think that would be a good idea, when in the future a new release of mlflow has issues and we have to roll-back to a previous version.

**Comment by zmhassan** [comment]
@4n4nd Yes. Thats correct.

**Pull request merged**